projects/clober/index.js

```diff
@@ -1,13 +1,11 @@
 const abi = require("./abi.json");
+const { fetchURL } = require('../helper/utils')
 const { sumTokens2 } = require('../helper/unwrapLPs')
 
-const FACTORY = "0x93A43391978BFC0bc708d5f55b0Abe7A9ede1B91"
-
 async function tvl(_, _b, _cb, { api }) {
-    let tokenAddresses = await api.fetchList({  lengthAbi: abi.nonce, itemAbi: abi.computeTokenAddress, target: FACTORY}) 
-    tokenAddresses = tokenAddresses.flat()
-    const markets = await api.multiCall({  abi: abi.market, calls: tokenAddresses })
-    const base = await api.multiCall({  abi: abi.baseToken, calls: markets}) 
+    const chainId = await api.getChainId()
+    const markets = (await fetchURL(`https://prod.clober-api.com/${chainId}/markets`)).data.markets.map((market) => market.address)
```

Comment:
I am not a fan of relying on external apis as they to break in the future, is there some alternative for this? like fetching via event logs?

Reply:
@g1nt0ki https://github.com/DefiLlama/DefiLlama-Adapters/pull/6194/commits/6a3848cd1f5e712dfdfefb9f9fa04f7aa2cbef66

Good! I've fixed it!